tools/klee/main.cpp

```diff
@@ -1587,9 +1652,47 @@ int main(int argc, char **argv, char **envp) {
         << handler->getNumPathsExplored() << ", among which\n";
   stats << "KLEE: done:     early-terminating paths (instruction time limit, solver timeout, max-depth reached) = "
         << handler->getEarlyTermination() << "\n";
+
+  if (INTERPOLATION_ENABLED) {
+    stats << "KLEE: done:     average depth of completed paths = "
+          << (double)(handler->getTotalBranchingDepthOnExitTermination() +
+                      handler->getTotalBranchingDepthOnEarlyTermination() +
+                      handler->getTotalBranchingDepthOnErrorTermination()) /
+                 (double)(handler->getExitTermination() +
+                          handler->getEarlyTermination() +
+                          handler->getErrorTermination()) << "\n";
+    if (handler->getSubsumedTermination() == 0.0) {
+      stats << "KLEE: done:     average depth of subsumption paths = " << 0
+            << "\n";
+    } else {
+      stats << "KLEE: done:     average depth of subsumption paths = "
+            << (double)handler->getTotalBranchingDepthOnSubsumption() /
+                   (double)handler->getSubsumedTermination() << "\n";
+    }
+  }
+
+  if (INTERPOLATION_ENABLED) {
+    stats << "KLEE: done:     average instructions of completed paths = "
+          << (double)(handler->getTotalInstructionsDepthOnExitTermination() +
+                      handler->getTotalInstructionsDepthOnEarlyTermination() +
+                      handler->getTotalInstructionsDepthOnErrorTermination()) /
+                 (double)(handler->getExitTermination() +
+                          handler->getEarlyTermination() +
+                          handler->getErrorTermination()) << "\n";
+    if (handler->getSubsumedTermination() == 0.0) {
+      stats << "KLEE: done:     average instructions of subsumption paths = "
+            << 0 << "\n";
+    } else {
+      stats << "KLEE: done:     average instructions of subsumption paths = "
+            << (double)
+                   handler->getTotalInstructionPathsExploredOnSubsumption() /
+                   (double)handler->getSubsumedTermination() << "\n";
+    }
+  }
+
   if (INTERPOLATION_ENABLED)
     stats << "KLEE: done:     subsumed paths = "
-          << handler->getSubsumptionTermination() << "\n";
+          << handler->getSubsumedTermination() << "\n";
```

Comment:
I think "subsumed paths" is good for the message, but for the method name, `getSubsumptionTermination` reads better because "subsumption termination" has a clearer meaning than "subsumed termination".